tests/connector/tests/hydrator_api_test.go

```diff
@@ -93,13 +98,26 @@ func TestHydrators(t *testing.T) {
 				}
 			}
 
-			//then
 			assert.Equal(t, true, hasAuth)
+
+			// check if the gql resolver is not sending used/expired auths
+			if testCase.clientType == "Application" {
+				appSystemAuths = fixtures.GetApplication(t, ctx, directorClient.DexGraphqlClient, cfg.Tenant, appID).Auths
+				assert.Len(t, appSystemAuths, 0)
+			}
 		})
 
 		t.Run("should resolve certificate for "+testCase.clientType, func(t *testing.T) {
 			//given
-			token, err := testCase.tokenGenerationFunc(t, testCase.clientId)
+			var err error
+			var token externalschema.Token
+
+			if testCase.clientType == "Application" {
```

Comment:
As Acho pointed out this snippets can be simplified (like it was before) as they are identical in each test case